### PR TITLE
fix(tflite): match PyTorch predict() preprocessing and mask decoder

### PR DESCRIPTION
**Pull request by omkar-334**
## What does this PR do?

The TFLite inference helper produced inputs the PyTorch graph never saw, and decoded masks with a different half-pixel convention than `PostProcess.forward`. After this change, FP32 TFLite is bit-exact to PyTorch for both object detection and segmentation. FP16 and dynamic-range int8 are within IoU 0.965 at production thresholds with zero class mismatches.

The conversion graph itself was never broken: a same-input probe shows boxes/logits `max|diff| <= 0.0001` between PyTorch and the existing FP32 `.tflite`. Everything we were seeing came from the inference-side pre/post-processing being slightly off from the training-time pipeline.

### Changes
Two helpers added to `inference.py` to fix where the TFLite predict path diverges from PyTorch's.

`_preprocess_image` (extracted from `_run_inference`). The old code called `PIL.Image.resize` with no filter argument. PIL's default has been BICUBIC since Pillow 9.1, but `RFDETR.predict` resizes with BILINEAR via `torchvision.transforms.functional.resize`. Same image, different input tensor, model sees something it wasn't trained on. The new helper uses `torchvision.transforms.functional` (`to_tensor → resize → normalize`) when torch is importable for byte-for-byte parity, and falls back to `PIL.Image.resize` with explicit BILINEAR for torch-free deployments (still close to PyTorch, ~0.016 max residual).

`_decode_masks` (rewritten) + `_bilinear_resize_half_pixel` (new fallback). The old `_decode_masks` did `PIL.Image.fromarray(..., "F").resize(..., BILINEAR)` per mask. PIL's BILINEAR uses a corner-aligned half-pixel convention; PyTorch's `F.interpolate(..., mode="bilinear", align_corners=False)` (what `PostProcess.forward` calls) uses centered half-pixel. The shift is sub-pixel, but on thin or border-hugging masks it flipped enough pixels to drop mask IoU to ~0.5 even when the source logits matched. The rewrite calls `torch.nn.functional.interpolate` directly when torch is importable, and falls back to `_bilinear_resize_half_pixel` — a vectorised numpy implementation of centered half-pixel bilinear — when it isn't.

### Before vs after (apples-to-apples at threshold 0.3)

Object detection (RFDETRNano on dog/people-walking/bus/zidane, 43 detections total):

| Mode     | Metric           | Before    | After     | Note                                     |
| -------- | ---------------- | --------- | --------- | ---------------------------------------- |
| FP32     | IoU_worst        | 0.640     | **1.000** | the dog image was the regression source  |
| FP32     | conf_max_diff    | 0.129     | **0.000** |                                          |
| FP32     | class_mismatches | 1         | **0**     |                                          |
| FP16     | IoU_worst        | **0.406** | 0.996     | the IoU_worst FP16 disaster              |
| FP16     | conf_max_diff    | 0.167     | 0.012     |                                          |
| FP16     | class_mismatches | 2         | 0         |                                          |
| dyn_int8 | IoU_mean         | 0.979     | 0.99      | small lift; int8 was already weight-only |

Related to #1024

## Type of Change

- Bug fix (non-breaking change that fixes an issue)

## Testing

- [x] I have tested this change locally
- [ ] I have added/updated tests for this change

### CODE REVIEW

File: src/rfdetr/export/_tflite/inference.py [hunk 1]
```diff
@@ -71,12 +71,41 @@ def _create_interpreter(model_path: str | Path) -> Any:
...
+    Uses the half-pixel center convention so mask upsampling matches ``PostProcess.forward``.
+
+    Args:
+        src: Source array of shape ``(K, src_h, src_w)``.
+        out_h: Target height in pixels.
+        out_w: Target width in pixels.
+
+    Returns:
+        Float32 array of shape ``(K, out_h, out_w)``.
+    """
```
[the remark this answers is not in the document]
Comment (omkar-334) [reply src/rfdetr/export/_tflite/inference.py:93]: Added a test for this

File: src/rfdetr/export/_tflite/inference.py [hunk 2]
```diff
@@ -94,12 +123,65 @@ def _decode_masks(mask_logits: NDArray[Any], out_size: tuple[int, int]) -> NDArr
...
         )
     width, height = out_size
-    out = np.empty((mask_logits.shape[0], height, width), dtype=np.bool_)
-    for i, logit_map in enumerate(mask_logits):
-        mask_img = PILImage.fromarray(logit_map.astype(np.float32), mode="F")
-        resized = mask_img.resize((width, height), _PIL_BILINEAR)
-        out[i] = np.asarray(resized) > 0.0
-    return out
+    resized = _bilinear_resize_half_pixel(mask_logits.astype(np.float32), height, width)
+    return resized > 0.0
```
[the remark this answers is not in the document]
Comment (omkar-334) [reply src/rfdetr/export/_tflite/inference.py:152]: torch is now prioritized. fixed

**Review by Borda — APPROVED** [review]
